### PR TITLE
Offset API for 'scroll to rule' behaviour 

### PR DESCRIPTION
**Pull request by jonathonherbert**
## What does this change?

This PR adds an offset API for the 'scroll to rule' sidebar behaviour. The example implementation centres the rules in the editor.

scroll-to-match

## Implementation notes

Because we don't know much about our consumer's editor, we defer to them with a `getScrollOffset` callback.

It's a callback, rather than a static value, because the editor size may change during the page lifecycle (for example, if the user alters the window size), and it's likely we'd need to adjust accordingly to e.g. centre matches correctly.

## How to test

In the example implementation, click a rule in the sidebar in a long-ish document. The editor should scroll to the match in the document, centering it on the page.

## How can we measure success?

Plugin consumers have more control about how the plugin scrolls matches into view.

### CODE REVIEW

File: src/ts/utils/decoration.ts [hunk 1]
```diff
@@ -201,3 +201,7 @@ export const findSingleDecoration = (
   }
   return decorations[0];
 };
+
+export const maybeGetDecorationElement = (matchId: string): HTMLElement | null => document.querySelector(
```
Comment (jonathonherbert) [inline src/ts/utils/decoration.ts:205]: I figure it's better to colocate this with the decoration code.

File: pages/index.ts [hunk 2]
```diff
@@ -50,6 +50,10 @@ if (editorElement && sidebarNode) {
     })
   });
 
+  // When the user scrolls to matches, place the match in the middle of the editor.
+  const menuHeight = 47;
```
Comment (SHession) [inline pages/index.ts:54]: Where does this come from /Is there a nicer way to get this value?

Comment (SHession) [reply pages/index.ts:54]: I withdraw the question, as it's only related to the demo implementation.

Comment (jonathonherbert) [reply pages/index.ts:54]: Haha, yeah, that was my thought 😁

**Pull request merged**